proposals/lambda-discard-parameters.md

```diff
@@ -0,0 +1,20 @@
+# Lambda discard parameters
+
+## Summary
+
+Allow discards (`_`) to be used as parameters of lambdas and anonymous methods.
+For example:
+- lambdas: `(_, _) => 0`, `(int _, int _) => 0`
+- anonymous methods: `delegate(int _, int _) { return 0; }`
+
+## Motivation
+
+Unused parameters do not need to be named. The intent of discards is clear, i.e. they are unused/discarded.
+
+## Detailed design
+
+In the parameter list of a lambda or anonymous method with more than one parameter named `_`, such parameters are discard parameters.
+Note: if a single parameter is named `_` then it is a regular parameter for backwards compatibility reasons.
+
+Discard parameters do not introduce any names to any scopes.
```

Comment:
>introduce any names to any scopes [](start = 26, length = 33)

Please identify the intended change to 8.7 **Scopes**.

You will need to modify the name lookup rules to cause this intent to be reified in the specification.  For example, we have the following text in ECMA's **12.7.3 Simple Names** (giving the meaning of a simple name as an expression):

> - If `K` is zero and the *simple-name* appears within a *block* and if the *block*’s (or an enclosing *block*’s) local variable declaration space (§8.3) contains a local variable, parameter or constant with name `I`, then the *simple-name* refers to that local variable, parameter or constant and is classified as a variable or value.

Note that the discard parameter is in the declaration space by 8.3, and is a parameter named `_` (see line 16 above), so without a change to the quoted paragraph it is indeed found during name lookup.  Please identify the intended changes to the specification.